presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestSingleDistinctAggregationToGroupBy.java

```diff
@@ -85,11 +73,14 @@ public void testDistinctWithFilter()
         tester().assertThat(new SingleDistinctAggregationToGroupBy())
                 .on(p -> p.aggregation(builder -> builder
                         .globalGrouping()
-                        .addAggregation(p.variable("output"), expression("count(DISTINCT input1) filter (where input2 > 0)"), ImmutableList.of(BIGINT))
-                        .source(
-                                p.values(
-                                        p.variable("input1"),
-                                        p.variable("input2")))))
+                        .source(p.values(p.variable("input1"), p.variable("input2")))
```

Comment:
now I see you've put source first everywhere. is there a reason?

Reply:
I wanted to call p.variable(...) first, so when I call p.rowExpression(...) it has registered the variables and can create the rowExpression. It will fail otherwise